### PR TITLE
Update CODING_STYLE.md

### PR DESCRIPTION
**Pull request by 3n16m4**
add 'east const' coding style rule and example

### CODE REVIEW

**Comment by christianparpart** [comment]
Hey. Many thanks. I will give details ASAP. See you soon.

File: docs/CODING_STYLE.md [hunk 1]
```diff
@@ -14,12 +14,19 @@
...
+- east const instead of west const
 
 ### Example
 
 ```cpp
 namespace org::coding_style::naming_conventions
 {
+    void eastConst() {
+        int const a = 42;        // a is const
+        int const *p = &a;       // value in p is const, p is not const.
```
Comment (christianparpart) [inline docs/CODING_STYLE.md:26]:
```suggestion
        int const* p = &a;       // value in p is const, p is not const.
```

File: docs/CODING_STYLE.md [hunk 2]
```diff
@@ -14,12 +14,19 @@
...
+        int const *const p = &a; // both value and p are const.
```
Comment (christianparpart) [inline docs/CODING_STYLE.md:27]:
```suggestion
        int const* const p = &a; // both value and p are const.
```

Comment (3n16m4) [reply docs/CODING_STYLE.md:27]: seems reasonable to me, the asterisk operator should be tied to the variable.

**Pull request merged**